spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/security/saml2/Saml2RelyingPartyPropertiesTests.java

```diff
@@ -87,6 +87,20 @@ void customizeSsoSignRequestsIsTrueByDefault() {
 				.isSignRequest()).isEqualTo(true);
 	}
 
+	@Test
+	void customizeLocalEntityIdTemplate() {
+		bind("spring.security.saml2.relyingparty.registration.simplesamlphp.local-entity-id-template",
+				"{baseUrl}/saml2/local-entity-id");
+		assertThat(this.properties.getRegistration().get("simplesamlphp").getLocalEntityIdTemplate())
+				.isEqualTo("{baseUrl}/saml2/local-entity-id");
+	}
+
+	@Test
+	void customizeLocalEntityIdTemplateDefaultsToServiceProviderMetadata() {
+		this.properties.getRegistration().put("simplesamlphp", new Saml2RelyingPartyProperties.Registration());
+		assertThat(this.properties.getRegistration().get("simplesamlphp").getLocalEntityIdTemplate()).isNull();
```

Comment:
If we change the `localEntityIdTemplate` property to have a default value of `{baseUrl}/saml2/service-provider-metadata/{registrationId}` this assertion will have to change to match. I would also be good to test that the default in the properties matches the default in the builder. That could be done with something like this:

```
assertThat(RelyingPartyRegistration.withRegistrationId("id")).extracting("localEntityIdTemplate")
		.isEqualTo(new Saml2RelyingPartyProperties.Registration().getLocalEntityIdTemplate());
```